File: src/framework/controllers/auth/forgot-password-submit.ts
import { IRequest, IResponse } from "../../../common/types";
import { z } from "zod";
import { Controller, HttpMethod } from "../types";
import databases from "../../../data/data-sources/injection";

export class ForgotPasswordSubmitController implements Controller<void> {
  method = HttpMethod.POST;

  route = "/auth/forgotPasswordSubmit";

  validator = z.object({
    email: z.string().trim(),
    code: z.string().length(6),
    newPassword: z.string().min(8),
  });

  execute(req: IRequest<z.infer<typeof this.validator>>, res: IResponse<void>) {
    return databases().authDatabase.forgotPasswordSubmit(
      req.body.email,
      req.body.code,
      req.body.newPassword
    );
  }
}

File: lib/src/data/backend/auth/forgot_password_submit.dart
part of backend;

extension ForgotPasswordSubmit on _Auth {
  Future<Failure?> forgotPasswordSubmit({
    required String email,
    required String code,
    required String newPassword,
  }) =>
      _request(
        route: '/auth/forgotPasswordSubmit',
        method: HttpMethod.post,
        data: {
          'email': email,
          'code': code,
          'newPassword': newPassword,
        },
      );
}
